### PR TITLE
Feature: System resources were not exposed

### PR DESCRIPTION
**Pull request by hoh**
The scheduling of persistent VMs requires external services to fetch the available system resources of the host.

Solution: Add a new HTTP endpoint on `/about/usage/system` that exposes system resources and system properties of the host machine.

### CODE REVIEW

File: vm_supervisor/resources.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+
+class CpuUsage(BaseModel):
+    count: int
+    load_average: LoadAverage
+    core_frequencies: CoreFrequencies
+
+
+class MemoryUsage(BaseModel):
+    total_kB: int
```
Comment (odesenfans) [inline vm_supervisor/resources.py:49]: Nitpick: not a fan of the case choice here, maybe `total_kb`?

File: vm_supervisor/resources.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    load1: float
+    load5: float
+    load15: float
+
+    @classmethod
+    def from_psutil(cls, psutil_loadavg: Tuple[float, float, float]):
+        return cls(load1=psutil_loadavg[0],
+                   load5=psutil_loadavg[1],
+                   load15=psutil_loadavg[2],
+                   )
```
Comment (odesenfans) [inline vm_supervisor/resources.py:28]: Formatting: run black?

File: vm_supervisor/resources.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        return self.available_kB / 1000
+
+
+class UsagePeriod(BaseModel):
+    start_timestamp: datetime
+    duration_seconds: float
+
+    class Config:
+        json_encoders = {
+            datetime: lambda v: v.isoformat(),
```
Comment (odesenfans) [inline vm_supervisor/resources.py:72]: That's not the default?

File: vm_supervisor/resources.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            load_average=LoadAverage.from_psutil(psutil.getloadavg()),
+            core_frequencies=CoreFrequencies.from_psutil(psutil.cpu_freq()),
+        ),
+        mem=MemoryUsage(
+            total_kB=psutil.virtual_memory().total / 1000,
+            available_kB=psutil.virtual_memory().available / 1000,
+        ),
+        disk=DiskUsage(
+            total_kB=psutil.disk_usage(settings.PERSISTENT_VOLUMES_DIR).total
+            // 1000,  # 10 GB,
```
Comment (odesenfans) [inline vm_supervisor/resources.py:120]: Why "10G" and "9GB" as comments?

File: vm_supervisor/resources.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    def available_MB(self) -> float:
+        return self.available_kB / 1000
+
+
+class DiskUsage(BaseModel):
+    total_kB: int
+    available_kB: int
+
+    @property
+    def available_MB(self) -> float:
```
Comment (odesenfans) [inline vm_supervisor/resources.py:62]: The property is unused for both models.

File: vm_supervisor/resources.py (hunk 1 above)
Comment (hoh) [reply vm_supervisor/resources.py:49]: Uppercase is the symbol for `byte`, lowercase for `bit`. These should not be confused.

File: vm_supervisor/resources.py (hunk 4 above)
Comment (hoh) [reply vm_supervisor/resources.py:120]: Test artifacts (I was overriding the values in tests), should not be merged.

**Comment by odesenfans** [comment]
Btw, isn't this the same PR as #194 ?

File: vm_supervisor/resources.py (hunk 1 above)
Comment (odesenfans) [reply vm_supervisor/resources.py:50]: True that.

**Review by odesenfans — APPROVED** [review]

**Pull request merged**